pytheus/backends.py

```diff
@@ -1,17 +1,26 @@
-from abc import ABC, abstractmethod
 import importlib
 import json
 import os
+
+from abc import ABC, abstractmethod
 from threading import Lock
-from typing import Any, Tuple
+from typing import Any, TYPE_CHECKING
+
+import redis
 
 from pytheus.exceptions import InvalidBackendClassException, InvalidBackendConfigException
 
+
+if TYPE_CHECKING:
+    from pytheus.metrics import Metric
+
+
 BackendConfig = dict[str, Any]
 
 
 class Backend(ABC):
-    def __init__(self, config: BackendConfig):
+    def __init__(self, config: BackendConfig, metric: 'Metric') -> None:
```

Comment:
currently passing the whole metric to retrieve the `name` & `_labels` information.
We can just pass what we need in the future maybe 🤔

Reply:
So this means both objects keep a reference of each other, right?

Reply:
correct!